selinux/qubes-core-qrexec.te

```diff
@@ -0,0 +1,47 @@
+policy_module(qubes-core-qrexec, 0.0.1)
+
+require {
+	type staff_t, init_t, initrc_t, sysadm_t, xauth_exec_t, local_login_t, tmpfs_t, qubes_var_run_t;
+	class passwd rootok;
+}
+
+### BEGIN UGLY HACK ###
+# There is no good way to use a custom label for a login program such as qrexec.
+# There are various bad ways, but they either don't always work or require
+# editing system files.  local_login_t is the best option available for the
+# qrexec agent, so use it.  Use aliases (defined here) so that this can easily
+# be fixed later.
+define(`qubes_qrexec_agent_t', `local_login_t')
+### END UGLY HACK ###
+
+# The attribute `qubes_qrexec_domain` determines which confined domains that are
+# allowed to use qrexec.
+attribute qubes_qrexec_domain;
+stream_connect_pattern(qubes_qrexec_domain, qubes_var_run_t, qubes_qrexec_socket_t, qubes_qrexec_agent_t)
+dev_rw_xen(qubes_qrexec_domain)
+allow qubes_qrexec_domain qubes_qrexec_socket_t:sock_file rw_sock_file_perms;
+
+# staff_t and sysadm_t can use qrexec.  Unprivileged user domains (such as user_t)
+# cannot.
```

Comment:
Does it also prevent normal user from using `qrexec-client-vm`, for example breaking qvm-copy? If so, it makes `user_t` mostly useless...

Reply:
This should probably be controlled by an SELinux boolean.  The intended purpose of this is to lock down `user_t` if a qube is e.g. a server with mutually distrusting users who log in over SSH.  Such users should not be allowed to use qrexec.  Up to you what the default should be.